src/nap.js

```diff
@@ -157,7 +157,9 @@ function bySelector(){
 }
 
 function wrap(fn, stack) {
-  return stack.reduce(middleware, fn)
+  return function wrapped() {
+    stack.reduce(middleware, fn).apply(null, arguments)
+  }
```

Comment:
> For what it's worth, I might just produce a word bomb on why I think middleware doesn't make any sense anyway – at least, in this incarnation I don't think they make sense – but I'll leave that in a different issue.

Would love to see that sometime :+1:. I've also moved on from the [middleware pattern for extensions in general to decorators](https://github.com/rijs/fullstack#contributing), which is still middleware, but more explicit, as beautifully explained in https://github.com/zeit/micro/issues/8.